reviewer/reviewer.go

```diff
@@ -0,0 +1,264 @@
+		allUsers := make(map[string]struct{})
+		allUsers = shoveIntoMap(allUsers, child.RequestedUsers)
+
+		if len(child.RequestedTeams) > 0 {
+			randomTeam := child.RequestedTeams[r.Intn(len(child.RequestedTeams))]
+			teamInfo := strings.Split(randomTeam, "/")
+			team, _, err := client.Teams.GetTeamBySlug(ctx, teamInfo[0], teamInfo[1])
+			if err != nil {
+				logger.Debug().Err(err).Msgf("Unable to get member listing for team %s", randomTeam)
+				//return nil, errors.Wrapf(err, "Unable to get information for team %s", randomTeam)
+			} else {
+				teamMembers, err := listAllTeamMembers(ctx, client, team)
+				if err != nil {
+					logger.Debug().Err(err).Msgf("Unable to get member listing for team %s", randomTeam)
+					//return nil, errors.Wrapf(err, "Unable to get member listing for team %s", randomTeam)
+				} else {
+					allUsers = shoveIntoMap(allUsers, teamMembers)
+				}
+			}
+		}
+
+		if len(child.RequestedOrganizations) > 0 {
+			randomOrg := child.RequestedOrganizations[r.Intn(len(child.RequestedOrganizations))]
+			orgMembers, err := listAllOrgMembers(ctx, client, randomOrg)
+			if err != nil {
+				return nil, errors.Wrapf(err, "Unable to get member listing for org %s", randomOrg)
+			}
+			allUsers = shoveIntoMap(allUsers, orgMembers)
+		}
+
+		allCollaborators, err := listAllCollaborators(ctx, client, prctx.RepositoryOwner(), prctx.RepositoryName())
+		if err != nil {
+			return nil, errors.Wrap(err, "Unable to get collaborators")
+		}
+		collaboratorPermissions := make(map[string]string)
+
+		if child.RequestedWriteCollaborators || child.RequestedAdmins {
+			for _, c := range allCollaborators {
+				perm, _, err := client.Repositories.GetPermissionLevel(ctx, prctx.RepositoryOwner(), prctx.RepositoryName(), c)
+				if err != nil {
+					return nil, errors.Wrapf(err, "failed to determine permission level of %s on repo %s", c, prctx.RepositoryName())
+				}
+				collaboratorPermissions[c] = perm.GetPermission()
+			}
+		}
+
+		if child.RequestedAdmins {
+			var repoAdmins []string
+			for _, c := range allCollaborators {
+				if collaboratorPermissions[c] == "admin" {
+					repoAdmins = append(repoAdmins, c)
+				}
+			}
+			allUsers = shoveIntoMap(allUsers, repoAdmins)
+		}
+
+		if child.RequestedWriteCollaborators {
+			var repoCollaborators []string
+			for _, c := range allCollaborators {
+				if collaboratorPermissions[c] == "write" {
+					repoCollaborators = append(repoCollaborators, c)
+				}
+			}
+			allUsers = shoveIntoMap(allUsers, repoCollaborators)
+		}
+
+		// Remove author before randomly selecting, since github will fail to assign _anyone_
+		if _, ok := allUsers[prctx.Author()]; ok {
+			delete(allUsers, prctx.Author())
+		}
+
+		// Remove any users who aren't collaborators, since github will fail to assign _anyone_
+		for k := range allUsers {
+			if collaboratorPerm, ok := collaboratorPermissions[k]; ok {
+				if collaboratorPerm != "admin" && collaboratorPerm != "write" {
```

Comment:
I know you must have write or higher for GitHub to count your review, but is it necessary to be requested? At least for teams, I thought granting a team read access was enough to allow assignment. Are users different?

This might be another reason to leave a comment on the PR. Since policy-bot will accept reviews from a larger range of people than GitHub, we'll have to decide if we should arbitrarily limit the pool by using review requests.

Reply:
I think I can remove this check as incorrect. The real check is just if the user shows up in the collaborator map, and if not, to remove them.